internal/painter/gl/draw.go

```diff
@@ -98,11 +97,102 @@ func (p *painter) drawRaster(img *canvas.Raster, pos fyne.Position, frame fyne.S
 }
 
 func (p *painter) drawRectangle(rect *canvas.Rectangle, pos fyne.Position, frame fyne.Size) {
-	if (rect.FillColor == color.Transparent || rect.FillColor == nil) && (rect.StrokeColor == color.Transparent || rect.StrokeColor == nil || rect.StrokeWidth == 0) {
-		return
```

Comment:
Is there any specific reason as to why this was removed?

Reply:
I think it was the old func that was deleted and written from scratch for GL. The rectangle is used very often and this if-statement is called repeatedly during rendering and makes the execution slower. I would say the developer sees on the screen if the rectangle is visible and would change the code. Is there an advantage of this if-statement?

Reply:
I think the intent is to have transparent rectangles (and similar) not be rendered at all and thus cost basically nothing. Let's check with the others before adding it back though.

Reply:
An if statement like that is not going to cost anything, but indeed save unnecessary transfer to the gpu. It actually improve things.

Reply:
I put the if statement a call stack higher and fixed double type assertion. Later a func could manage this color and stroke checks because the circle shape would need it too.
```Go
	case *canvas.Rectangle:
		if (obj.FillColor == color.Transparent || obj.FillColor == nil) && (obj.StrokeColor == color.Transparent || obj.StrokeColor == nil || obj.StrokeWidth == 0) {
			return
		}
		if obj.Radius == 0 {
			p.drawRectangle(obj, pos, frame)
		} else {
			p.drawRoundRectangle(obj, pos, frame)
		}
```